src/fastapi_app/dependencies.py

```diff
@@ -88,10 +86,11 @@ async def get_azure_credentials() -> azure.identity.DefaultAzureCredential | azu
         raise e
 
 
+azure_credentials = get_azure_credentials()
```

Comment:
I usually try to avoid globally setting any variable immediately in a file, as it often makes testing harder (since mocks must be setup before that file gets imported). However, it appears tests are passing, so maybe it's okay in this case. We could move to a function updates a global if it ends up being problematic, I suppose.

Reply:
we are mocking the original module that's why it's no issue for any Default_azure_credentials

Reply:
I just understood why you did what you did with the life span thingy when I was tackling this, seems like a way more complicated issue than that. 😆